sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -872,6 +873,15 @@ object JdbcUtils extends Logging {
     // E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
     val sql = s"CREATE TABLE $tableName ($strSchema) $createTableOptions"
     executeStatement(conn, options, sql)
+    if (options.tableComment.nonEmpty) {
+      try {
+        executeStatement(
+          conn, options, dialect.getTableCommentQuery(tableName, options.tableComment))
```

Comment:
instead of having a dialect API to set table comment, can we have a dialect API to create table?

Reply:
Yes, we can have a dialect API to create table. The benefit is that I can append table comment in CREATE TABLE for MySQL, but I think I need to change ```logWarning``` to public so I can log a Waring in the dialects if table comment is not supported.